src/register.jl

```diff
@@ -649,8 +649,9 @@ function register(
         # branch registry repo
         @debug("branch registry repo")
         git = gitcmd(registry_path, gitconfig)
-        run(pipeline(`$git checkout -f master`; stdout=devnull))
-        if branch != "master"
+        default_branch = splitpath(String(readchomp(`git symbolic-ref refs/remotes/origin/HEAD`)))[end]
```

Comment:
Use `$git` to get the prepared `git` invocation with configuration parameters.
`String` is redundant.
None of my registries has a `refs/remotes/origin/HEAD`, so this doesn't seem very robust.
It also makes the assumption that you have a remote called `origin`, which is no more guaranteed than the default branch being called `master`. (Not a regression though.)

Reply:
> Use `$git` to get the prepared `git` invocation with configuration parameters.
:+1: will update
> `String` is redundant.
`readchomp` seems to return a `SubString` rather than a `String`, so I ran into a method error without the explicit `String`. I was running on Julia 1.2, but I see that it has been fixed on later versions. However, since the compat bounds for `RegistryTools` have Julia 1.1, I think we still need the `String`
> None of my registries has a `refs/remotes/origin/HEAD`, so this doesn't seem very robust.
Yes, that was a worry for me. I think any cloned repo should have a remote, but I wasn't sure what to do in cases where users don't host their remote somewhere. Do you have any suggestion there?
> It also makes the assumption that you have a remote called `origin`, which is no more guaranteed than the default branch being called `master`. (Not a regression though.)
Yes, that's true, hadn't considered that. I think this assumption is less likely to be a problem in practice as there's an active effort among many organizations to deprecate the use of the term `master`, and I know of no such effort to move away from `origin`. Would appreciate suggestions for other ways to detect the main branch - I haven't come across any yet

Reply:
> readchomp seems to return a SubString rather than a String, so I ran into a method error without the explicit String. I was running on Julia 1.2, but I see that it has been fixed on later versions. However, since the compat bounds for RegistryTools have Julia 1.1, I think we still need the String.

Hah, I've forgot that `splitpath` was ever that limited. (Apparently wasn't fixed until 1.4.)

> Yes, that was a worry for me. I think any cloned repo should have a remote, but I wasn't sure what to do in cases where users don't host their remote somewhere. Do you have any suggestion there?

I do have a `refs/remotes/origin` but no `HEAD` inside there.

These are parts of RegistryTools that I don't use in my package, so really not very familiar with it, but it looks like it's supposed to use a clone that is created by `get_registry`, which reduces the variability of what can go there, possibly making `origin` a non-problem. To me it seems like the most robust way to find out the default branch is to just see what you get when you clone, then making sure to return to it every time you have done things with the clone.